sql_generators/glean_usage/glean_app_ping_views.py

```diff
@@ -66,26 +66,105 @@ def generate_per_app(self, project_id, app_info, output_dir=None):
+                    print(
+                        f"Cannot UNION 'moz-fx-data-shared-prod.{app_dataset}.{view_name}': {e}"
+                    )
+                    print("Generate view only for release channel.")
+
+                    unioned_schema = Schema.for_table(
+                        "moz-fx-data-shared-prod",
+                        release_app["document_namespace"].replace("-", "_"),
+                        view_name,
+                        partitioned_by="submission_timestamp",
+                    )
+                    release_only = True
+
+            # generate the SELECT expression used for UNIONing the stable tables;
+            # fields that are not part of a table, but exist in others, are set to NULL
+            datasets = []
+            for app in app_info if not release_only else [release_app]:
+                app_dataset = app["document_namespace"].replace("-", "_")
+
+                # compare table schema with unioned schema to determine fields that need to be NULL
+                schema = Schema.for_table(
+                    "moz-fx-data-shared-prod",
+                    app_dataset,
+                    view_name,
+                    partitioned_by="submission_timestamp",
+                )
+
+                if schema.schema["fields"] == []:
+                    # check for empty schemas (e.g. restricted ones) and skip for now
+                    continue
+
+                select_expression = self._generate_select_expression(
+                    unioned_schema.schema["fields"], schema.schema["fields"]
+                )
+
+                datasets.append(
+                    dict(
+                        select_expression=select_expression,
+                        dataset=app_dataset,
+                        table=view_name,
+                    )
+                )
+
+            if datasets == []:
+                # nothing to render
+                continue
+
+            # render view SQL
+            render_kwargs = dict(
+                project_id=project_id, target_view=full_view_id, datasets=datasets
+            )
+            rendered_view = reformat(view_template.render(**render_kwargs))
+
+            # write generated SQL files to destination folders
             if output_dir:
+                underlying_view_id = ".".join(
+                    [
+                        "moz-fx-data-shared-prod",
+                        repo["app_id"].replace("-", "_"),
+                        view_name,
+                    ]
+                )
+
                 write_sql(
                     output_dir,
                     full_view_id,
                     "view.sql",
-                    reformat(
-                        VIEW_QUERY_TEMPLATE.format(
-                            full_view_id=full_view_id,
-                            target=underlying_view_id,
-                        )
-                    ),
+                    reformat(rendered_view),
```

Comment:
reformat is already applied to rendered_view
```suggestion
                    rendered_view,
```